{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "46a17ddf-3935-4854-8221-279e0fbbb666",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.10/site-packages/datasets/load.py:1461: FutureWarning: The repository for mc4 contains custom code which must be executed to correctly load the dataset. You can inspect the repository content at https://hf.co/datasets/mc4\n",
      "You can avoid this message in future by passing the argument `trust_remote_code=True`.\n",
      "Passing `trust_remote_code=True` will be mandatory to load this dataset from the next major release of `datasets`.\n",
      "  warnings.warn(\n",
      "Downloading builder script: 100%|██████████| 9.68k/9.68k [00:00<00:00, 12.8MB/s]\n",
      "Downloading readme: 100%|██████████████████| 16.0k/16.0k [00:00<00:00, 18.8MB/s]\n",
      "/root/.cache/huggingface/modules/datasets_modules/datasets/mc4/78f7a2b7e2524fa44ee464ef429d011c365f5fe129283869e7fd76856aacb83a/mc4.py:284: FutureWarning: Dataset 'mc4' is deprecated and will be deleted. Use 'allenai/c4' instead.\n",
      "  warnings.warn(\n",
      "data/path_list/CC-MAIN-2023-50\n",
      "  0%|                                                    | 0/10 [00:00<?, ?it/s]downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00100.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00100.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00100.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00100.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00100.warc\n",
      "downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00101.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00101.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00101.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00101.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00101.warc\n",
      "downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00102.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00102.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00102.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00102.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00102.warc\n",
      "downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00103.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00103.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00103.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00103.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00103.warc\n",
      "downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00104.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00104.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00104.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00104.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00104.warc\n",
      " 50%|██████████████████████                      | 5/10 [00:00<00:00, 49.82it/s]downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00105.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00105.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00105.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00105.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00105.warc\n",
      "downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00106.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00106.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00106.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00106.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00106.warc\n",
      "downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00107.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00107.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00107.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00107.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00107.warc\n",
      "downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00108.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00108.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00108.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00108.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00108.warc\n",
      "downloading mode with s3 \n",
      "download from s3://commoncrawl/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00109.warc.gz to data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00109.warc\n",
      "Error loading SSO Token: Token for vomuuv does not exist\n",
      "fail loading https://data.commoncrawl.org/crawl-data/CC-MAIN-2023-50/segments/1700679099514.72/warc/CC-MAIN-20231128115347-20231128145347-00109.warc.gz\n",
      "parsing data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00109.warc\n",
      "fail loading data/warc/crawl-data_CC-MAIN-2023-50_segments_1700679099514.72_warc_CC-MAIN-20231128115347-20231128145347-00109.warc\n",
      "100%|███████████████████████████████████████████| 10/10 [00:00<00:00, 83.93it/s]\n",
      "Traceback (most recent call last):\n",
      "  File \"/app/warc/download_and_parse.py\", line 157, in <module>\n",
      "    main()\n",
      "  File \"/app/warc/download_and_parse.py\", line 154, in main\n",
      "    put_s3(f\"{submit_dir}.zip\", f\"{submit_dir}.zip\")\n",
      "  File \"/app/warc/src/s3_utils.py\", line 46, in put_s3\n",
      "    s3.upload_file(file_name, ZIP_UPLOAD_S3_BUCKET, file_path)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/boto3/s3/inject.py\", line 145, in upload_file\n",
      "    return transfer.upload_file(\n",
      "  File \"/usr/local/lib/python3.10/site-packages/boto3/s3/transfer.py\", line 371, in upload_file\n",
      "    future.result()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/s3transfer/futures.py\", line 103, in result\n",
      "    return self._coordinator.result()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/s3transfer/futures.py\", line 266, in result\n",
      "    raise self._exception\n",
      "  File \"/usr/local/lib/python3.10/site-packages/s3transfer/tasks.py\", line 139, in __call__\n",
      "    return self._execute_main(kwargs)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/s3transfer/tasks.py\", line 162, in _execute_main\n",
      "    return_value = self._main(**kwargs)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/s3transfer/upload.py\", line 764, in _main\n",
      "    client.put_object(Bucket=bucket, Key=key, Body=body, **extra_args)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/client.py\", line 553, in _api_call\n",
      "    return self._make_api_call(operation_name, kwargs)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/client.py\", line 989, in _make_api_call\n",
      "    http, parsed_response = self._make_request(\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/client.py\", line 1015, in _make_request\n",
      "    return self._endpoint.make_request(operation_model, request_dict)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/endpoint.py\", line 119, in make_request\n",
      "    return self._send_request(request_dict, operation_model)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/endpoint.py\", line 198, in _send_request\n",
      "    request = self.create_request(request_dict, operation_model)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/endpoint.py\", line 134, in create_request\n",
      "    self._event_emitter.emit(\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/hooks.py\", line 412, in emit\n",
      "    return self._emitter.emit(aliased_event_name, **kwargs)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/hooks.py\", line 256, in emit\n",
      "    return self._emit(event_name, kwargs)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/hooks.py\", line 239, in _emit\n",
      "    response = handler(**kwargs)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/signers.py\", line 105, in handler\n",
      "    return self.sign(operation_name, request)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/signers.py\", line 186, in sign\n",
      "    auth = self.get_auth_instance(**kwargs)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/signers.py\", line 301, in get_auth_instance\n",
      "    frozen_credentials = credentials.get_frozen_credentials()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/credentials.py\", line 634, in get_frozen_credentials\n",
      "    self._refresh()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/credentials.py\", line 522, in _refresh\n",
      "    self._protected_refresh(is_mandatory=is_mandatory_refresh)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/credentials.py\", line 538, in _protected_refresh\n",
      "    metadata = self._refresh_using()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/credentials.py\", line 685, in fetch_credentials\n",
      "    return self._get_cached_credentials()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/credentials.py\", line 695, in _get_cached_credentials\n",
      "    response = self._get_credentials()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/credentials.py\", line 2150, in _get_credentials\n",
      "    token = initial_token_data.get_frozen_token().token\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/tokens.py\", line 82, in get_frozen_token\n",
      "    self._refresh()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/tokens.py\", line 95, in _refresh\n",
      "    self._protected_refresh()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/tokens.py\", line 109, in _protected_refresh\n",
      "    self._frozen_token = self._refresh_using()\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/tokens.py\", line 306, in _refresher\n",
      "    token_dict = self._token_loader(start_url, session_name=session_name)\n",
      "  File \"/usr/local/lib/python3.10/site-packages/botocore/utils.py\", line 3365, in __call__\n",
      "    raise SSOTokenLoadError(error_msg=error_msg)\n",
      "botocore.exceptions.SSOTokenLoadError: Error loading SSO Token: Token for vomuuv does not exist\n"
     ]
    }
   ],
   "source": [
    "! python download_and_parse.py 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2dafbcb-d99d-432e-b3a8-79da2663538a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
